client/components/dashboard/dashboard-toolbar-directive_test.js

```diff
@@ -0,0 +1,150 @@
+  var dashboardService;
+
+  const explorer = p3rf.perfkit.explorer;
+
+  beforeEach(module('explorer'));
+  beforeEach(module('p3rf.perfkit.explorer.templates'));
+
+  beforeEach(inject(function(
+      _$rootScope_, _$compile_, _$timeout_, _dashboardService_) {
+    scope = _$rootScope_.$new();
+    $compile = _$compile_;
+    $timeout = _$timeout_;
+
+    dashboardService = _dashboardService_;
+  }));
+
+  describe('compilation', function() {
+
+    it('should succeed as a standalone element.', function() {
+      function compile() {
+        var directiveElement = angular.element(
+            '<dashboard-toolbar />');
+
+        $compile(directiveElement)(scope);
+        scope.$digest();
+      }
+      expect(compile).not.toThrow();
+    });
+  });
+
+  describe('should contain an element to', function() {
+    var actualElement, actualController;
+
+    beforeEach(inject(function() {
+      actualElement = angular.element('<dashboard-toolbar />');
+      $compile(actualElement)(scope);
+
+      scope.$digest();
+
+      actualController = actualElement.controller('dashboardToolbar');
+    }));
+
+    it('create a new dashboard', function() {
+      var targetElement = actualElement.find(
+          'button.dashboard-create');
+      expect(targetElement.length).toBe(1);
+
+      spyOn(actualController, 'createDashboard');
+      targetElement.click();
+      expect(actualController.createDashboard).toHaveBeenCalled();
+    });
+
+    it('show the open dashboard options', function() {
+      var targetElement = actualElement.find(
+          'button.dashboard-open');
+      expect(targetElement.length).toBe(1);
+    });
+
+    it('list the open dashboard options', function() {
+      var targetElement = actualElement.find(
+          'ul.dashboard-open-dropdown');
+      expect(targetElement.length).toBe(1);
+    });
+
+    it('open the dashboard admin page', function() {
+      var targetElement = actualElement.find(
+          'li.dashboard-open-admin');
+      expect(targetElement.length).toBe(1);
+
+      spyOn(actualController, 'openDashboardAdmin');
+      targetElement.click();
+      expect(actualController.openDashboardAdmin).toHaveBeenCalled();
+    });
+
+    it('save the current dashboard', function() {
+      var targetElement = actualElement.find(
+          'button.dashboard-save');
+      expect(targetElement.length).toBe(1);
+
+      spyOn(dashboardService, 'saveDashboard');
+      targetElement.click();
+      expect(dashboardService.saveDashboard).toHaveBeenCalled();
+    });
+
+    it('save a copy of the current dashboard', function() {
+      var targetElement = actualElement.find(
+          'button.dashboard-copy');
+      expect(targetElement.length).toBe(1);
+
+      spyOn(actualController, 'saveDashboardCopy');
+      targetElement.click();
+      expect(actualController.saveDashboardCopy).toHaveBeenCalled();
+    });
+
+    it('download the current dashboard', function() {
```

Comment:
The describe() for this block is "should contain an element to", so "download" is the correct tense here.